### ISSUE TITLE
OCP 4.11.5 install failure: no terraform binary

### ISSUE THREAD

**Issue by thbkrkr** [elastic/cloud-on-k8s#6041]
OCP 4.11.5 installation (recently added in #6029) fails with:

`[2022-09-27T00:49:24.857Z] level=error msg=failed to fetch Cluster: failed to generate asset "Cluster": failure applying terraform for "cluster" stage: failed to create cluster: failed doing terraform init: fork/exec /home/ocp4-ci-1111790071683/terraform/bin/terraform: no such file or directory`

It looks like a bug. Terraform binaries are managed by the openshift-installer. I see in the [changelog](https://mirror.openshift.com/pub/openshift-v4/clients/ocp/4.11.5/changelog.html) that there was a change to move the binaries to the install directory instead of /tmp.

- https://issues.redhat.com/browse/CORS-1952
- https://github.com/openshift/installer/pull/5715
- https://github.com/openshift/installer/pull/5810
- https://github.com/openshift/installer/pull/5825

**Comment by barkbay** [comment]
> It looks like a bug.

I managed to create a cluster from the command line, I think the bug is in the deployer or the Docker image.

**Comment by barkbay** [comment]
The `terraform` binary is not statically linked, we are however running the installer in a [`scratch` container](https://github.com/elastic/cloud-on-k8s/blob/41edbb93f99fd3b5f413512380bb0f8cba032656/hack/deployer/clients/ocp/Dockerfile#L15). I think it is the root cause of the problem:

```
/ # ls -la /home/barkbay-ocp2458979669/terraform/bin/terraform
-rwxr-xr-x    1 root     root      55111680 Sep 28 13:24 /home/barkbay-ocp2458979669/terraform/bin/terraform
```

```
/ # /home/barkbay-ocp2458979669/terraform/bin/terraform
/bin/sh: /home/barkbay-ocp2458979669/terraform/bin/terraform: not found
```

```
ldd /home/barkbay-ocp2458979669/terraform/bin/terraform
	linux-vdso.so.1
	libpthread.so.0
	libc.so.6 
	/lib64/ld-linux-x86-64.so.2
```

**Comment by barkbay** [comment]
```
readelf -a /home/barkbay-ocp2458979669/terraform/bin/terraform  | grep NEEDED
 0x0000000000000001 (NEEDED)             Shared library: [libpthread.so.0]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
```

**Comment by thbkrkr** [comment]
Yes, I just came to the same conclusion. Let's change the base image.
